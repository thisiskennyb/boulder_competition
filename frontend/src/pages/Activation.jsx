import { useParams } from "react-router-dom"
import { useEffect } from "react"
import axios from 'axios'
import { useNavigate } from 'react-router-dom';
import { toast } from "react-toastify";



export default function Activation() {

    const {uid, token} = useParams()
    const navigate = useNavigate()

    useEffect(() => {
        const activateAccount = async () => {
            try {
                const response = await axios.post("http://localhost/api/users/activation/", {
                    uid: uid,
                    token: token
                });

                // Handle the response as needed
                toast.success('Account activated');
                navigate('/login')
            } catch (error) {
                // Handle errors
                toast.error('Whoops something went wrong');
                navigate('/')
            }
        };

        // Call the activation function
        activateAccount();
    }, [uid, token]);
    
    return (
        <h1 className="text-teal-400">Activation Page</h1>
    )
}