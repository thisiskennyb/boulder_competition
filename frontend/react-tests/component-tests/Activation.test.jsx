import {expect, test} from "vitest"
import Activation from "../../src/pages/Activation"

test("Activation renders without crashing", () => {
        expect(<Activation/>).toMatchSnapshot()
    })
