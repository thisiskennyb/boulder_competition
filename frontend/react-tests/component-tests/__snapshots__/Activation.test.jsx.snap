// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Activation renders without crashing 1`] = `<Activation />`;
